webExtensionApis/study/schema.yaml

```diff
@@ -0,0 +1,164 @@
+# Interface schema
+# https://firefox-source-docs.mozilla.org/toolkit/components/extensions/webextensions/schema.html
+# possible types are here
+# - https://searchfox.org/mozilla-central/source/toolkit/components/extensions/Schemas.jsm#2418-2426
+#
+# - properties
+# - functions
+# - events
+# - types
+#
+# - type property:
+#   boolean, integer, number, string, array, object, or any.
+#
+
+# TODO
+# - add json schema types
+# - do we want to expose any of the sampling / hashing stuff?
+# - mention / decide how to handle all
+---
+- namespace: study
+  description: Interface for shield studies, for use by `background.js` scripts.
+  apiVersion: 5  # for dev use
+  types:
+  - name: studySetup
+  - name: weightedVariations
+  - name: anEnding
+  - name: telemetrySelectionOptions
+  - name: prefTypes
+
+  functions:
+  - name: configure
+    type: function
+    description: Configure the study.  Most things can't work without this
+    async: true
+    parameters:
+    - name: studySetup
+      type: object
+      # with a variation included.
+      # expiration?
+
+
+  - name: simpleDeterministicVariation
+    returns: Name of a variation from weightedVariations
+    async: true
+    parameters:
+    - name: weightedVariations
+      type: object
+    - name: fraction
+      optional: true
+      type: Number
+
+  - name: endStudy
+    parameters:
+    - name: anEnding
+    - type: object
+    # should be anEnding Object
+    returns: none
+
+  ## informational things
+  - name: permissions
+    async: true
+    # I don't like this name!
+    # TODO: should there be an arg to just ask for pioneer?  shield?
+    returns: object of permissions with keys shield, pioneer, telemetry, 'ok'
+  - name: userInfo
```

Comment:
Not sure that this method needs to be part of the public API? Perhaps as an internal method to append query strings to the survey URL...